subsys/bluetooth/audio/ascs.c

```diff
@@ -1378,6 +1378,68 @@ static int ase_config(struct bt_ascs *ascs, struct bt_ascs_ase *ase,
 	return 0;
 }
 
+int bt_audio_unicast_server_config_ase(struct bt_conn *conn,
+		struct bt_audio_stream *stream, struct bt_codec *codec,
+		const struct bt_codec_qos_pref *qos_pref)
+{
+	struct bt_ascs *ascs;
+	struct bt_ascs_ase *ase;
+	struct bt_audio_ep *ep;
+
+	CHECKIF(conn == NULL || stream == NULL || codec == NULL ||
+		qos_pref == NULL) {
+		LOG_DBG("NULL value(s) supplied)");
+		return -EINVAL;
+	}
+
+	ascs = ascs_get(conn);
+	ep = stream->ep;
+
+	if (ep == NULL) {
+		sys_snode_t *ase_node;
+		uint8_t ase_next_id = 1;
+
+		SYS_SLIST_FOR_EACH_NODE(&ascs->ases, ase_node) {
+			ase_next_id++;
+		}
+
+		ase = ase_get(ascs, ase_next_id);
+		if (ase == NULL) {
+			LOG_WRN("No free ASE found for config ASE ID 0x%02x", ase_next_id);
+			return -ENOTSUP;
+		}
```

Comment:
This function shall configure a new ASE. 

If there already is an ASE with ID 1, then `ase_get` returns that one, which is not what we want. Instead what we should do is call `ase_find` until that returns NULL, which indicates an unused ASE. Once we then have the "free index", we can call `ase_new` with that to get a new unused ASE.